node/cmd/ccq/query_server.go

```diff
@@ -209,7 +219,21 @@ func runQueryServer(cmd *cobra.Command, args []string) {
 	signal.Notify(sigterm, syscall.SIGTERM)
 	go func() {
 		<-sigterm
-		logger.Info("Received sigterm. exiting.")
+		if statServer != nil && *shutdownDelay1 != 0 {
+			logger.Info("Received sigterm. disabling health checks and pausing.")
+			statServer.disableHealth()
+			time.Sleep(time.Duration(*shutdownDelay1) * time.Second)
+			logger.Info("Waiting for outstanding requests to complete before shutting down.")
+			for count := 0; count < int(*shutdownDelay2); count++ {
+				time.Sleep(time.Second)
+				if pendingResponses.Empty() {
+					break
+				}
+			}
+			logger.Info("Done waiting. shutting down.")
```

Comment:
should we warn here if the requests aren't empty when we start shutting down anyway?